### ISSUE TITLE
Touching disabled Wacom pad interferes with pressure/width of line

### ISSUE THREAD

**Issue by Svalorzen** [mypaint/mypaint#1083]
#### Description of the problem

This is a relatively old bug I think since I've had it for a long time, but until today I thought it was a problem with my system. I am on Lubuntu 20.04, and I use a Wacom Intuos 3 (PTZ-431W).

When drawing lines at low pressures, if I accidentally touch the pad stripe on the tablet the line spikes in width (at the end of the line I am purposefully increasing the pressure to show that touching the pad has no major effect there):

![image](https://user-images.githubusercontent.com/1609228/83129481-2bb94480-a0dd-11ea-87a6-ac9658138daa.png)

In theory, I have disabled the pad both with `xsetwacom` (touch/gesture are set to off), and from MyPaint itself, as I have set it to ignore the pad. I am pretty sure this is what is causing the problem because if set MyPaint to *not* ignore the pad, this is what happens:

![image](https://user-images.githubusercontent.com/1609228/83129687-78048480-a0dd-11ea-9959-9c8e048afc48.png)

Which is much worse. All other applications do not read the pad without special settings; with GIMP this problem does not appear. It could be that this problem is related to #879 , but I am not sure.

This is unfortunately especially annoying for me as I am left-handed, so my palm tends to go over the strip pretty much all the time..

I am adding here all settings read from the tablet; maybe there's something else I could change and I did make a mistake in setting it up:

```
Property 'Wacom Tablet Area' does not exist on device.
Button: Actions are not supported by xorg.conf. Try shell format (-s) instead.
Button: Actions are not supported by xorg.conf. Try shell format (-s) instead.
Button: Actions are not supported by xorg.conf. Try shell format (-s) instead.
Button: Actions are not supported by xorg.conf. Try shell format (-s) instead.
Option "DebugLevel" "0"
Option "CommonDBG" "0"
Option "Suppress" "2"
Option "RawSample" "4"
Property 'Wacom Pressurecurve' does not exist on device.
Option "Mode" "Absolute"
Property 'Wacom Hover Click' does not exist on device.
Option "Touch" "off"
Property 'Wacom Hardware Touch Switch' does not exist on device.
Option "Gesture" "off"
Option "ZoomDistance" "0"
Option "ScrollDistance" "0"
Option "TapTime" "250"
Property 'Wacom Proximity Threshold' does not exist on device.
Property 'Wacom Rotation' does not exist on device.
'RelWheelUp' requires exactly 0 value(s).
'RelWheelDown' requires exactly 0 value(s).
'AbsWheelUp' requires exactly 0 value(s).
'AbsWheelDown' requires exactly 0 value(s).
'AbsWheel2Up' requires exactly 0 value(s).
'AbsWheel2Down' requires exactly 0 value(s).
'StripLeftUp' requires exactly 0 value(s).
'StripLeftDown' requires exactly 0 value(s).
'StripRightUp' requires exactly 0 value(s).
'StripRightDown' requires exactly 0 value(s).
Option "Threshold" "0"
Option "Serial" "0"
Property 'Wacom Pressure Recalibration' does not exist on device.
Option "PanScrollThreshold" "2600"
```

#### Basic system details

MyPaint version, installed from apt:

2.0.0+gitexport.af5225a
(Python 3.8.2, GTK 3.24.18, GdkPixbuf 2.40.0, Cairo 1.16.0, GLib 2.64.1)

Operating system: Lubuntu 20.04
Desktop environment: LXQT/Openbox

#### Steps to reproduce

1. Start MyPaint
2. Draw lines lightly
3. Use the other hand to repeatedly scroll the pad

**Comment by jplloyd** [comment]
Hi,

I don't think the xsetwacom settings you mentioned are disabling the strip - since the model in question supports neither touch nor gestures, disabling either would have no effect (according to the xsetwacom man page they should already be disabled by default).

Run `xev` from a terminal, place the cursor in the little window and use the stripe to see if events _are_ generated.

No matter what, if MyPaint still reacts to those events despite being set to ignore that device, it _may_ very well be a bug.

**Comment by Svalorzen** [comment]
The strip does not seem to record any particular additional event when I touch the strip. Not sure what this means. The output of xev for the pen itself also doesn't seem to show any particular alterations when the strip is touched, but then again pressure is not shown as well, so it might be a limitation of xev?

I'm not sure what the property of the strip is if not touch; to activate it you have to slide over it with your finger. I have a script that disables a bunch of properties of the strip; without it I can for example scroll my browser by passing my finger on it, and after disabling it this doesn't work; so I am assuming that the settings I am applying do change something.

**Comment by jplloyd** [comment]
Alright, then my assumption was incorrect. I thought that touch/gestures would only refer to touch input on the tablet area itself on newer devices, and that touch strips/wheels would be considered to be part of the pad.

So with your current setup, other programs don't react to the strip at all (it doesn't generate a bunch of keypresses)? Which version of GIMP did you test this with? If it works fine with a recent version of GIMP that would suggest we're doing something wrong with the input handling.

**Comment by Svalorzen** [comment]
GIMP is 2.10.18.

Other programs don't seem to react to the strip at all, no. I mean, if MyPaint is, this means something is going through, but I'm unsure to test exactly what that is.

I checked again my "disabling" script, and you are right, it doesn't touch "touch" nor "gesture"; it disables "StripLeftUp" and "StripLeftDown". If I manually re-enable them, then applications start perceiving the strip as clicks (GIMP, terminal starts selecting text, etc). When I disable them, this stops. MyPaint however is still able to feel the pad for some reason. Could be that it is reading another separate property?

**Comment by jplloyd** [comment]
Definitely fishy. I will see if I can reproduce this with the wheel on my Intuos 4. I also know someone who still uses a variant of Intuos 3, so if I can't get the same behaviour with the wheel I may be able to see if there's something special about those models.

**Comment by Svalorzen** [comment]
Nice thanks! If you think of anything I could do to help you pinpoint this let me know :)

**Comment by jplloyd** [comment]
So I can replicate this partially with the wheel, but disabling the pad in the preferences (setting it to either "ignore" or "non-painting tasks only" fixes the stroke breaks, as expected.

The way I "disabled" the wheel was to map the motions to "button 0". Does your script "disable" the strip in the same way?

**Comment by Svalorzen** [comment]
No, I mostly disable the two settings I mentioned plus all random buttons as I am clumsy and I end up clicking them as I move. For clarity I'm just going to paste here the script as to not keep you guessing :)

```
#!/bin/bash

STYLUSID=$(xsetwacom --list | grep -i stylus | cut -f 2 | cut -f 2 -d ' ')
PADID=$(   xsetwacom --list | grep -i pad    | cut -f 2 | cut -f 2 -d ' ')
ERASERID=$(xsetwacom --list | grep -i eraser | cut -f 2 | cut -f 2 -d ' ')
CURSORID=$(xsetwacom --list | grep -i cursor | cut -f 2 | cut -f 2 -d ' ')

echo "Using stylus id = $STYLUSID"
echo "Using pad id = $PADID"
echo "Using eraser id = $ERASERID"
echo "Using cursor id = $CURSORID"

if [ -z "$STYLUSID" ] || [ -z "$PADID" ] || [ -z "ERASERID" ] || [ -z "CURSORID" ]; then
    echo "Could not extract some ids, aborting.."
    exit
fi

# Remove strip zooming (pad)
xsetwacom --set "$PADID" StripLeftUp 0
xsetwacom --set "$PADID" StripLeftDown 0
# Remove tablet buttons (pad)
for i in {1..8}; do xsetwacom --set "$PADID" Button $i 0; done
# Remove pen buttons but main (stylus)
for i in {2..9}; do xsetwacom --set "$STYLUSID" Button $i 0; done
# Remove eraser buttons but main (eraser)
for i in {2..9}; do xsetwacom --set "$ERASERID" Button $i 0; done
```

I'm not sure how to map the motions to button 0; please do let me know if you think I should try it.

**Comment by jplloyd** [comment]
Thank you.

Replace this
```
xsetwacom --set "$PADID" StripLeftUp 0
xsetwacom --set "$PADID" StripLeftDown 0
```
with this
```
xsetwacom --set "$PADID" StripLeftUp "Button 0"
xsetwacom --set "$PADID" StripLeftDown "Button 0"
```

and see if that fixes it. I still find it weird that the input is not completely ignored when set to ignore the pad.

**Comment by Svalorzen** [comment]
Unfortunately, nothing has changed..

**Comment by Svalorzen** [comment]
Ok, some news. First I tried to disable the "right pad" settings and "wheel" settings as well, since they were there and they could have been interfering. These are my current settings for the pad:

```
Property 'Wacom Tablet Area' does not exist on device.
xsetwacom set "13" "Button" "1" "button +0 "
xsetwacom set "13" "Button" "2" "button +0 "
xsetwacom set "13" "Button" "3" "button +0 "
xsetwacom set "13" "Button" "8" "button +0 "
xsetwacom set "13" "ToolDebugLevel" "0"
xsetwacom set "13" "TabletDebugLevel" "0"
xsetwacom set "13" "Suppress" "2"
xsetwacom set "13" "RawSample" "4"
Property 'Wacom Pressurecurve' does not exist on device.
xsetwacom set "13" "Mode" "Absolute"
Property 'Wacom Hover Click' does not exist on device.
xsetwacom set "13" "Touch" "off"
Property 'Wacom Hardware Touch Switch' does not exist on device.
xsetwacom set "13" "Gesture" "off"
xsetwacom set "13" "ZoomDistance" "0"
xsetwacom set "13" "ScrollDistance" "0"
xsetwacom set "13" "TapTime" "250"
Property 'Wacom Proximity Threshold' does not exist on device.
Property 'Wacom Rotation' does not exist on device.
xsetwacom set "13" "RelWheelUp" "1" "button +0 "
xsetwacom set "13" "RelWheelDown" "2" "button +0 "
xsetwacom set "13" "AbsWheelUp" "3" "button +0 "
xsetwacom set "13" "AbsWheelDown" "4" "button +0 "
xsetwacom set "13" "AbsWheel2Up" "5" "button +0 "
xsetwacom set "13" "AbsWheel2Down" "6" "button +0 "
xsetwacom set "13" "StripLeftUp" "1" "button +0 "
xsetwacom set "13" "StripLeftDown" "2" "button +0 "
xsetwacom set "13" "StripRightUp" "3" "button +0 "
xsetwacom set "13" "StripRightDown" "4" "button +0 "
xsetwacom set "13" "Threshold" "0"
xsetwacom set "13" "BindToSerial" "0"
Property 'Wacom Pressure Recalibration' does not exist on device.
xsetwacom set "13" "PanScrollThreshold" "2600"
```

Now, while playing around a bit more, I discovered that this is also triggered by the physical buttons! Each click to any of the physical buttons on the tablet creates a little "dot" while drawing. But they should also be disabled!

**Comment by jplloyd** [comment]
Good info, definitely seems like the device is not really ignored then. 

I was actually going to install lubuntu 20.04 on an old laptop anyway, so I will revisit this in a few days to check if it's a driver issue, or maybe (more likely) use of deprecated interfaces on our end. If I can't fully replicate it there either it may have to wait until I can test this with an Intuos3 (hopefully within 3-4 weeks).

A temporary workaround would be to invert the mapping of the tablet area so you can flip the tablet around and have the buttons on the right side, but I guess with that model such an arrangement would be pretty inconvenient (because of the cable) and uncomfortable. Making a sleeve of some kind to cover the pad/strip might be a better workaround.

**Comment by Svalorzen** [comment]
Yeah, it's not super urgent, I can definitely cover up the strip. 

Thanks a lot for the quick responses!

**Comment by jplloyd** [comment]
Quick update on this:

I could replicate the behavior on a fresh lubuntu 20.04 install, and when testing on ubuntu 16.04 again I could replicate it there as well. I must have been too hasty in my previous test (unless the bug is tablet-borne :P ).

I still need to figure out exactly why this is happening and how to solve it, but a better workaround than the previous suggestions is to just disable the pad device entirely in your setup script:

```
xinput --disable $PADID
```

**Comment by Svalorzen** [comment]
This works beautifully, thanks! I'll keep following the issue in case you need more feedback from me :)
